md/tests/test_views.py

```diff
@@ -0,0 +1,63 @@
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from md.models import Agency, Stop
+
+
+class ViewTests(TestCase):
+    def tearDown(self):
+        Stop.objects.all().delete()
+        Agency.objects.all().delete()
```

Comment:
I think this is the real fix for the need to explicitly delete objects after each testcase (from https://docs.djangoproject.com/en/1.9/topics/testing/tools/#multi-database-support):

```
diff --git md/tests/test_api.py md/tests/test_api.py
index 2d2386f..1f425fd 100644
--- md/tests/test_api.py
+++ md/tests/test_api.py
@@ -12,6 +12,8 @@ from md.tests import factories
 
 class AgencyTests(APITestCase):
 
+    multi_db = True
+
     def test_list_agencies(self):
         """Test Agency list"""
         agency = factories.AgencyFactory()
diff --git md/tests/test_normalization.py md/tests/test_normalization.py
index 6bba0f7..640816e 100644
--- md/tests/test_normalization.py
+++ md/tests/test_normalization.py
@@ -13,6 +13,8 @@ from md.models import UNKNOWN_PURPOSE
 
 class TestFieldNormalization(TestCase):
 
+    # if db calls are added: multi_db = True
+
     def test_ethnicity(self):
         data = (
             ('WHITE', 'W'),
diff --git md/tests/test_views.py md/tests/test_views.py
index a701fd1..6bf713b 100644
--- md/tests/test_views.py
+++ md/tests/test_views.py
@@ -4,20 +4,22 @@ from md.models import Agency, Stop
 
 
 class ViewTests(TestCase):
-    def tearDown(self):
-        Stop.objects.all().delete()
-        Agency.objects.all().delete()
+
+    multi_db = True
 
     def test_home(self):
```

(and similar for the nc tests)

I still get one md test failure when the query cache is enabled for tests.  (I didn't test nc with this changes.)  Maybe the Foo.objects.all().delete() code did something to bypass the cache on subsequent queries?  Do you get an error?

```
======================================================================
FAIL: test_agency_list_sorted_agencies (md.tests.test_views.ViewTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/trawick/caktus/git/Traffic-Stops-foo/md/tests/test_views.py", line 52, in test_agency_list_sorted_agencies
    self.assertEqual(3, len(sorted_agencies))
AssertionError: 3 != 1
```